.github/workflows/golang.yaml

```diff
@@ -47,7 +45,7 @@ jobs:
         - name: Install Go
           uses: actions/setup-go@v5
           with:
-            go-version: '1.20'
+            go-version-file: 'go.mod'
```

Comment:
I would recommend doing something like https://github.com/NVIDIA/k8s-device-plugin/blob/47d1aa131da1ddc017e4b1e2c8133690e1bd94f8/.github/workflows/golang.yaml#L33-L41 here.

We want to ensure that we're using consistent go versions.